arangod/Aql/Query.h

```diff
@@ -324,6 +329,9 @@ class Query {
 
   /// @brief pointer to vocbase the query runs in
   TRI_vocbase_t& _vocbase;
+  
+  /// @brief transaction context to use for this query
+  std::shared_ptr<transaction::Context> _transactionContext;
```

Comment:
a transaction::Context is already part of Methods '_trx' why is it necessary to create another instance

Reply:
its optional so we can use the SmartContext